### PR TITLE
SNOW-1826001 Follow up with post-merge comments, remove coverage upload for AST doctest.

### PR DESCRIPTION
**Pull request by sfc-gh-lspiegelberg**
SNOW-1826001

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [x] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

   Removes some deadcode which was removed in main. Removes uploading coverage for AST doctest.

### CODE REVIEW

File: .github/workflows/precommit.yml [hunk 1]
```diff
@@ -393,18 +393,6 @@ jobs:
           PYTEST_ADDOPTS: --color=yes --tb=short --enable_ast
           TOX_PARALLEL_NO_SPINNER: 1
         shell: bash
-      - name: Combine coverages
-        run: python -m tox -e coverage --skip-missing-interpreters false
```
Comment (sfc-gh-aalam) [inline .github/workflows/precommit.yml:397]: why is this removed?

Comment (sfc-gh-lspiegelberg) [reply .github/workflows/precommit.yml:397]: It looked to me that the coverage of this gate conflicts with the existing doctest coverage, some path conflict. Removing it to ensure smoother gates (seemed something flakey).

Comment (sfc-gh-lspiegelberg) [reply .github/workflows/precommit.yml:397]: Example here: https://github.com/snowflakedb/snowpark-python/actions/runs/12039250410/job/33566836571

**Review by sfc-gh-aalam — APPROVED** [review]

**Review by sfc-gh-vbudati — APPROVED** [review]
